app/src/main/java/com/ataulm/whatsnext/di/AppComponent.kt

```diff
@@ -1,40 +1,30 @@
-package com.ataulm.whatsnext.di;
+package com.ataulm.whatsnext.di
 
 import android.app.Application
-import com.ataulm.letterboxd.LetterboxdApi
-import com.ataulm.letterboxd.auth.LetterboxdAuthApi
-import com.ataulm.whatsnext.AuthRepository
-import com.ataulm.whatsnext.BuildConfig
-import com.ataulm.whatsnext.LocalTokensStorage
+import com.ataulm.letterboxd.LetterboxdComponent
+import com.ataulm.letterboxd.LetterboxdRepository
 import com.ataulm.whatsnext.WhatsNextRepository
 import com.ataulm.whatsnext.account.SignInUseCase
 import com.ataulm.whatsnext.account.UserIsSignedInUseCase
 import com.ataulm.whatsnext.api.FilmConverter
 import com.ataulm.whatsnext.api.FilmRelationshipConverter
 import com.ataulm.whatsnext.api.FilmStatsConverter
 import com.ataulm.whatsnext.api.FilmSummaryConverter
-import com.ataulm.whatsnext.api.auth.AddAuthorizationInterceptor
-import com.ataulm.whatsnext.api.auth.LetterboxdAuthRepository
-import com.ataulm.whatsnext.api.auth.LetterboxdAuthenticator
-import com.chuckerteam.chucker.api.ChuckerInterceptor
 import dagger.BindsInstance
 import dagger.Component
 import dagger.Module
 import dagger.Provides
-import okhttp3.OkHttpClient
-import okhttp3.logging.HttpLoggingInterceptor
-import retrofit2.Retrofit
-import retrofit2.converter.moshi.MoshiConverterFactory
 import javax.inject.Singleton
 
-private const val LETTERBOXD_BASE_URL = "https://api.letterboxd.com/api/v0/"
-
 @Component(
+    dependencies = [
+        LetterboxdComponent::class
+    ],
     modules = [
         AppModule::class
     ]
 )
-@Singleton
+@AppSingletonScope
```

Comment:
There's two rules for components:

1. An un-scoped component cannot depend on scoped components.
2. A scoped component cannot depend on a component with the same scope.

Because of the second rule, we can't have `LetterboxdComponent` and the `AppComponent` both using `@Singleton`. So I introduced the `@Letterboxd` scope which is internal to the `letterboxd` module.

What I didn't understand is that if I leave this as `@Singleton` (and keep `@Letterboxd` for the other one) it also complains:

```
error: This @Singleton component cannot depend on scoped components:
@dagger.Component(dependencies = {com.ataulm.letterboxd.LetterboxdComponent.class}, modules = {com.ataulm.whatsnext.di.AppModule.class})
```

Is the `@Singleton` scope treated in a special way in Dagger? I guess I could have used `@Singleton` in `LetterboxdComponent` and only changed this one too (I didn't try).